### ISSUE TITLE
Export statement before variable initialization don't work with parcel

### ISSUE THREAD

**Issue by GrosSacASac** [parcel-bundler/parcel#716]
🐛 bug report

Export statement before variable initialization don't work with parcel

### 🎛 Configuration

no babel rc used

```js
{

  "devDependencies": {
    "babel-preset-env": "^1.6.1",
    "parcel-bundler": "^1.5.1"
  },
}
```

### 🤔 Expected Behavior
```

export {strings};

const strings = [
    `Hello Parcel`,
    `zzzzzzzzz`
];
```

 Hello World Hello Parcel
should work

### 😯 Current Behavior

 ReferenceError: can't access lexical declaration `strings' before initialization

Hello World only

### 💁 Possible Solution

make it work like in the browser

meanwhile put export at the end

### 🔦 Context

trying to bundle a working project to make it smaller. It already works without parcel

### 💻 Code Sample

See attachement

### 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel          | 1.5.1 
| Node          |   9 
| npm/Yarn      |   5.6
| Operating System | win

to try it in firefox es modules have to be enabled

### Reproduce

open zip, open parcel-test.html without parcel (with static file server ) it works

npm install

npm run parceltest

does not 

[parcel-test.zip](https://github.com/parcel-bundler/parcel/files/1682777/parcel-test.zip)

**Comment by fathyb** [comment]
Isn't Babel supposed to take care of this?

**Comment by davidnagli** [comment]
Is this even supposed to work? Does JavaScript allow you to export something before you define it??

**Comment by DeMoorJasper** [comment]
This is a babel issue, but this is more an issue of invalid JavaScript

**Comment by brandon93s** [comment]
> "no babel rc used"

Parcel needs the config file to understand what presets you want to use. Try adding a rc to your project. 

If babel is handling this, it seems coincidental to how the code is mangled. Or, it's "hoisting" the const for you. Either way, the provided code intentionally throws in ES6 and would ideally be restructured to be valid.

**Comment by brandon93s** [comment]
@GrosSacASac let us know if you still have trouble after adding a babel config.

**Comment by GrosSacASac** [comment]
I don't think my case needs babel at all. The attachments I provided works out of the box perfectly fine, in Opera and Firefox.

Maybe theses are not spec compliant ? Can anyone confirm ?

I read a bit about module specification, and I understood the following: When a module is parsed, 

 * first a list of export strings are found, then live bindings are made, with live variables initially set to undefined
 * then imports are executed,  then the js text is parsed again (fully this time) and executed

From what I understand the syntax I provided is valid and above mentioned browsers are behaving per spec.

But with parcel when , `parcel parcel-test.html` is used there is an error, on excution in the same browser
